### ISSUE TITLE
Enhance Custom Element Handling

### ISSUE THREAD

**Issue by mmcdole** [mmcdole/gofeed#205]
See the following: https://github.com/mmcdole/gofeed/issues/203#issuecomment-1483859708

I'm looking to enhance the custom element handling of gofeed so it can attempt to process arbitrary XML elements and make their data accessible.

**Comment by mmcdole** [comment]
Should I also consider handling custom elements within other elements?  Like, within an `<author>` element for example?  

Should each child struct have a customelement or map?

**Comment by mmcdole** [comment]
We also have our `extension` capability that parses arbitrary extensions.  

Perhaps we should get rid of the `CustomElement` idea and just tweak the extensions capability so it will parse anything, even non-namespaced extensions

https://github.com/mmcdole/gofeed/blob/master/rss/parser.go#L144

**Comment by rodmcelrath** [comment]
Yes!  This!Sent from my iPhoneOn Mar 25, 2023, at 5:08 PM, mmcdole ***@***.***> wrote:﻿
We also have our extension capability that parses arbitrary extensions.
Perhaps we should get rid of the CustomElement idea and just tweak the extensions capability so it will parse anything, even non-namespaced extensions
https://github.com/mmcdole/gofeed/blob/master/rss/parser.go#L144

**Comment by rodmcelrath** [comment]
Also possibly things that failed the valid rss test.  Example, had to pull a source from feed recently that was text only, no url.  Invalid rss so it didn’t come through.  Had to do a transformation on feed to make it look like an extension.  Then pass it through!Sent from my iPhoneOn Mar 25, 2023, at 8:29 PM, Rod McElrath ***@***.***> wrote:﻿Yes!  This!Sent from my iPhoneOn Mar 25, 2023, at 5:08 PM, mmcdole ***@***.***> wrote:﻿
We also have our extension capability that parses arbitrary extensions.
Perhaps we should get rid of the CustomElement idea and just tweak the extensions capability so it will parse anything, even non-namespaced extensions
https://github.com/mmcdole/gofeed/blob/master/rss/parser.go#L144

**Comment by rodmcelrath** [comment]
I get it. Not strictly valid RSS.  Source with no url which is required.  If you let it through some people would be surprised.But pubdate is supposed to be a standard format. Literally no one observes the standard. But the code bends over backward to get something valid.RSS standard is practically a joke. Some of the things I have seen!Sent from my iPhoneOn Mar 25, 2023, at 8:29 PM, Rod McElrath ***@***.***> wrote:﻿Yes!  This!Sent from my iPhoneOn Mar 25, 2023, at 5:08 PM, mmcdole ***@***.***> wrote:﻿
We also have our extension capability that parses arbitrary extensions.
Perhaps we should get rid of the CustomElement idea and just tweak the extensions capability so it will parse anything, even non-namespaced extensions
https://github.com/mmcdole/gofeed/blob/master/rss/parser.go#L144

**Comment by mmcdole** [comment]
This is the design discussion that became #276 (unify custom + extension into one arbitrary-element tree). Consolidating there.
